metricbeat/module/apache/status/status.go

```diff
@@ -87,10 +92,36 @@ func (m *MetricSet) Fetch(reporter mb.ReporterV2) error {
 	}
 
 	data, _ := eventMapping(scanner, m.Host())
+	event := mb.Event{
+		MetricSetFields: data,
+	}
 
-	if reported := reporter.Event(mb.Event{MetricSetFields: data}); !reported {
-		m.Logger().Error("error reporting event")
+	if m.isFleetMode {
+		event = adjustFleetEvent(event)
 	}
 
+	if reported := reporter.Event(event); !reported {
+		m.Logger().Error("error reporting event")
+	}
 	return nil
 }
+
+func adjustFleetEvent(event mb.Event) mb.Event {
+	var adjusted mb.Event
+	adjusted.MetricSetFields = event.MetricSetFields.Clone()
+
+	// Remove apache.hostname
+	adjusted.MetricSetFields.Delete("hostname")
```

Comment:
Actually this field should probably be removed when not running in fleet mode too. This is obtained from the config (using `m.Host()`), whose value will also end up in `service.address`.
https://github.com/elastic/beats/blob/af9788859a7ddd7df4b1677fc43b7533d5245f3b/metricbeat/module/apache/status/status.go#L94

Reply:
I left the hostname there because we need to ensure backward compatibility in Beats. Not sure if it's safe to remove it from everywhere.

Reply:
Yeah, it'd be breaking, let's leave this out of this PR.